### PR TITLE
Feature/get login page functioning

### PR DESCRIPTION
**Pull request by StevieAmb**
### Description

Include a summary of the change and which issue is fixed or feature was added

### Select One Thing
- [ ] This broke nothing
- [ ] This broke something
- [ ] This broke everything
- [x] This fixed something

### Fixes (if any)
 - This fixed the error handling by breaking up the method, so that the error handling was done in another method
 - This also did some edge casing so if the array kept coming up undefined it would get to the second if statement

### Features (if any)
 - Added the log in feature.
 -

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings/bugs
- [ ] I have checked variable and parameter names: are they descriptive and unique?
- [ ] Is the code *dry* (donot repeat yourself), SRP (single responsibility principle)?
- [ ] Is it easy for others to read?
- [ ] Is the syntax consistent? (e.g. use of semicolons, blank lines)
- [ ] Are commit messages formatted correctly?

### Next Steps:
 - I am going to fix the spacing between the drop down and the select button on the nav bar
 -

### CODE REVIEW

File: src/User.js [hunk 1]
```diff
@@ -51,18 +53,23 @@ class User {
...
-      return `Sorry for the inconvenience! Please try another room type!`
-    }
+      this.wantedRooms = wantedRooms;
   }
 
+  throwError = () => {
+    let desired = this.wantedRooms;
+    console.log("here", desired)
+    if(desired.length === 0) {
+       this.message = `I am SO SORRY! Please forgive me! I beg your pardon, and I promise if you pick another room, you can have one a tiny home. On me.`
```
Comment (Alynn022) [inline src/User.js:69]: Check sentence for grammar for error message, but otherwise, a very apologetic message to the customer!

File: src/scripts.js [hunk 2]
```diff
@@ -42,36 +42,57 @@ const errorHandlingLine = document.getElementById('handleMyError')
...
+function getData(userID) {
+  return Promise.all([fetchSingleUser(userID), fetchApiData('rooms'), fetchApiData('bookings')])
 }
 
+//on click
+const loginUser = (event) => {
+  event.preventDefault();
+  let username = usernameInput.value
+  let userID = username.substring(8)
+  if(userID > 0 && userID < 51 && passwordInput.value === 'overlook2021') {
```
Comment (Alynn022) [inline src/scripts.js:61]: Love that you added edge cases in case the user enters a userId that is not part of the data.

File: test/User-test.js [hunk 3]
```diff
@@ -72,20 +72,22 @@ describe('User', () => {
     it('Should be able to filter rooms based on room type', () => {
       user1.findBookingsByDate("2022/01/18", userBookingData)
       user1.findAvailableRooms(userRoomData);
-      let filteredRooms = user1.filterAvailableRooms('single room')
+      user1.filterAvailableRooms('single room')
 
-      expect(filteredRooms[0]).to.equal(userRoomData[2])
-      expect(filteredRooms[1]).to.equal(userRoomData[3])
-      expect(filteredRooms[2]).to.equal(userRoomData[4])
+      expect(user1.wantedRooms[0]).to.equal(userRoomData[2])
```
Comment (Alynn022) [inline test/User-test.js:77]: very thorough test for this it block.

**Review by Alynn022 — APPROVED** [review]
